java/client/src/main/java/glide/api/models/ArgsBuilder.java

```diff
@@ -0,0 +1,37 @@
+/** Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models;
+
+import com.google.protobuf.ByteString;
+import redis_request.RedisRequestOuterClass.Command.ArgsArray;
+
+// TODO docs for the god of docs
+public class ArgsBuilder {
+    ArgsArray.Builder commandArgs = null;
```

Comment:
We shouldn't build up ArgsArray always. We should use the ArgVecPointer when sizes are too large. See: #1708 .

Reply:
I agree that we have performance issue. But this change was done without informing people working on fixing the txn binary support. I am going to comment to large values tests and will circrle back to this after the GA. For now, correctness is `>>` performance for edge cases

Reply:
The performance change is a GA requirement. See: https://github.com/aws/glide-for-redis/issues/1726 .
The PR for this work was a port from a the Python implementation as wel (#1655).